### PR TITLE
Have nsupdate add expected quotation to TXT records

### LINKED ISSUE #43380: nsupdate handling record values with spaces
##### SUMMARY
When adding a record with values with spaces, for example a TXT record with SPF data like `"v=spf1 mx -all"` it will currently be added as `"v=spf1" "mx" "-all"`. This results in being finally parsed as  `v=spf1mx-all`, breaks the syntax of the record and I wouldn't consider this intuitive behavior.
As I found on https://stackoverflow.com/a/17255241 when double-quoting the record value like `'"v=spf1 mx -all"'` it will properly create the record.

If this is expected behavior it should be documented. Otherwise we should probably wrap it in the nsupdate module.

##### ISSUE TYPE
 - Documentation Report

##### COMPONENT NAME
nsupdate

##### ANSIBLE VERSION
```
2.7
```

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### STEPS TO REPRODUCE
```yaml
- name: "create spf txt record for {{ record }}"
  nsupdate:
    key_name: "{{ nsupdate_key_name }}"
    key_secret: "{{ nsupdate_key_secret }}"
    key_algorithm: "{{ nsupdate_key_algorithm }}"

    server: "{{ nsupdate_server | default('localhost') }}"
    state: "present"

    record: "{{ record }}"
    type: "txt"
    value: "v=spf1 mx -all"
    ttl: 300
```

##### EXPECTED RESULTS
TXT record with value `"v=spf1 mx -all"`

##### ACTUAL RESULTS
TXT record with value `"v=spf1" "mx" "-all"`

### PR DESCRIPTION
**Pull request by andreaso**
##### SUMMARY

This prevents the accidental creation of TXT records where every
single word gets split into its own string, such as TXT record values
in the format of `"foo" "bar" "baz"`. That being an implicit behavior
I have very hard to see anyone purposely relying on.

TXT record values can still explicitly be defined as one or more
strings, without any change in syntax.

Resolves #43380

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
nsupdate

##### ANSIBLE VERSION
```
ansible 2.7.0.dev0 (devel 551501f326) last updated 2018/07/28 14:49:34 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/andreas/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/andreas/labs/ansible/devel/lib/ansible
  executable location = /home/andreas/labs/ansible/devel/bin/ansible
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
```

### CODE REVIEW

File: lib/ansible/modules/net_tools/nsupdate.py [hunk 1]
```diff
@@ -207,8 +207,18 @@ def __init__(self, module):
         else:
             self.algorithm = module.params['key_algorithm']
 
+        if self.module.params['type'].lower() == 'txt':
```
Comment (Nothing4You) [inline lib/ansible/modules/net_tools/nsupdate.py:210]: This issue also affects SPF records (even though they are deprecated).
Maybe there are other record types aswell that use text values?

Comment (andreaso) [reply lib/ansible/modules/net_tools/nsupdate.py:210]: This is strictly a personal opinion, but I don't really feel inclined to add code to support a deprecated record type.

**Comment by andreaso** [comment]
@nerzhul: Thoughts?

**Comment by andreaso** [comment]
One week remaining until the Ansible 2.7 community modules freeze.

@nerzhul: Any opinion either way in regards to this PR?

**Comment by nerzhul** [comment]
shipit

**Pull request merged**